### PR TITLE
added podman documentation

### PR DESCRIPTION
**Pull request by hyunnaye**
Updated documentations by removing outdated versions and added podman documentation

Related issue/ticket: https://ucsc-cgl.atlassian.net/browse/DOCK-2236

### CODE REVIEW

File: docs/advanced-topics/docker-alternatives.rst [hunk 1]
```diff
@@ -73,6 +73,36 @@ add a line to your ``~/.dockstore/config``:
...
+    cwltool --podman <workflow> <input json>
+
+To set this option through Dockstore, add the following line to your ``~/.dockstore/config``:
+::
+
+    cwltool-extra-parameters: --podman
+
+Cromwell
+~~~~~~~~
+For cromwell, we can set a symbolic link to run Podman instead of docker on workflows.
```
Comment (coverbeck) [inline docs/advanced-topics/docker-alternatives.rst:99]: cromwell -> Cromwell, docker -> Docker

File: docs/advanced-topics/docker-alternatives.rst [hunk 2]
```diff
@@ -40,10 +40,10 @@ More information about compatibility of dockerfiles with Singularity
 can be found `here <https://sylabs.io/guides/3.4/user-guide/singularity_and_docker.html#best-practices>`__.
 
 Singularity can be installed following the instructions
-`here <https://sylabs.io/guides/3.4/user-guide/quick_start.html>`__. Note that the installation is relatively complicated
```
Comment (aofarrel) [inline docs/advanced-topics/docker-alternatives.rst:43]: Does this deletion mean that sudo no longer required?

File: docs/advanced-topics/docker-alternatives.rst [hunk 3]
```diff
@@ -40,10 +40,10 @@ More information about compatibility of dockerfiles with Singularity
...
-and requires ``sudo``. Neither the macOS version (Singularity Desktop) nor the Debian/Ubuntu package version currently
-available (2.6.1) is compatible. You will need to download a version >3.0.0 and build it from source.
+`here <https://github.com/sylabs/singularity/blob/main/INSTALL.md>`__.
 
+.. note:: Multiple installations of Go has been seen to cause a ``Go compiler not found`` error when installing Singularity.
```
Comment (aofarrel) [inline docs/advanced-topics/docker-alternatives.rst:45]: Consider replacing "has been seen" with "has been known"

File: docs/advanced-topics/docker-alternatives.rst [hunk 4]
```diff
@@ -73,6 +73,36 @@ add a line to your ``~/.dockstore/config``:
 
     cromwell-vm-options: -Dconfig.file=<absolute path to your Cromwell conf>
 
+Podman
+---------------
+
+Podman is a daemon-less alternative of Docker that allows end users to run docker/OCI containers without root privileges.
```
Comment (aofarrel) [inline docs/advanced-topics/docker-alternatives.rst:79]: We should probably define with daemon means in this context in the Dockstore dictionary.

Comment (aofarrel) [inline docs/advanced-topics/docker-alternatives.rst:79]: While we're at it, we should define OCI too. iirc this is the first time they're referenced in our docs.

File: docs/advanced-topics/docker-alternatives.rst [hunk 5]
```diff
@@ -73,6 +73,36 @@ add a line to your ``~/.dockstore/config``:
...
+    cwltool-extra-parameters: --podman
+
+Cromwell
+~~~~~~~~
+For cromwell, we can set a symbolic link to run Podman instead of docker on workflows.
+::
+
+    ln -s /usr/bin/podman /usr/bin/docker
+
+Note that docker should not be installed when running this command.
```
[the remark this answers is not in the document]
Comment (aofarrel) [reply docs/advanced-topics/docker-alternatives.rst:104]: I'd also reword this to "Note that you should not do this if you already have Docker installed on your machine."

File: docs/advanced-topics/docker-alternatives.rst [hunk 6]
```diff
@@ -73,6 +73,36 @@ add a line to your ``~/.dockstore/config``:
...
+
 Rootless Docker
 ---------------
```
Comment (aofarrel) [inline docs/advanced-topics/docker-alternatives.rst:108]: Is the stuff in this section still true? Either way, it's probably best to combine this section with the one you edited around line 19.

**Review by aofarrel — CHANGES_REQUESTED** [review]
Thanks for updating this. I think some clarifications are needed, but thanks for introducing me to podman!

File: docs/advanced-topics/docker-alternatives.rst (hunk 6 above)
Comment (hyunnaye) [reply docs/advanced-topics/docker-alternatives.rst:108]: I tested the other alternatives like Singularity and they all work with Docker installed. Except for podman as the symbolic link causes errors when Docker is installed

File: docs/advanced-topics/docker-alternatives.rst [hunk 7]
```diff
@@ -16,7 +16,7 @@ environment.
 
 Rootless Docker
 ---------------
-`Experiments <https://medium.com/@tonistiigi/experimenting-with-rootless-docker-416c9ad8c0d6#:~:text=Docker%20engine%20provides%20lots%20of,Linux%20you%20need%20privileged%20capabilities.>`__ with a version of Docker that does not require root permissions have been proposed and experimented with for a while now. In December 2020 with the release of Docker Engine v20.10, "rootless Docker" has officially become part of Docker itself. While this does open the door for some systems, it does require special setup and it has some limitations. You can read all about it on `Docker's own docs about Rootless Mode <https://docs.docker.com/engine/security/rootless/>`__.
```
Comment (aofarrel) [inline docs/advanced-topics/docker-alternatives.rst:19]: I still believe this section needs to be moved combined with the other rootless Docker section. Right now, this page's headers are:
* Rootless Docker
* Singularity
* Podman
* Rootless Docker, again

It makes more sense to just combine the two rootless docker sections.

File: docs/advanced-topics/docker-alternatives.rst [hunk 8]
```diff
@@ -16,7 +16,7 @@ environment.
```
Comment (aofarrel) [inline docs/advanced-topics/docker-alternatives.rst:16]: The paragraph that starts at line 7 doesn't mention podman, but it does mention singularity and rootless Docker. I would consider removing "including Singularity and rootless Docker" from that paragraph, and adding a very small table of contents under the paragraph that lists the alternatives we talk about (so people can quickly jump to the singularity section, the podman section, or the rootless Docker section).

File: docs/advanced-topics/docker-alternatives.rst [hunk 9]
```diff
@@ -73,6 +73,36 @@ add a line to your ``~/.dockstore/config``:
...
+
+cwltool
+~~~~~~~
+
+Podman is also available as a command line option for cwltool like this:
+::
+
+    cwltool --podman <workflow> <input json>
+
+To set this option through Dockstore, add the following line to your ``~/.dockstore/config``:
```
Comment (aofarrel) [inline docs/advanced-topics/docker-alternatives.rst:92]: Reword to "to set this as the default option when running via the Dockstore CLI" or something else that makes it clear this is for the Dockstore CLI, not Dockstore as a product/website/service (yes, this has confused people before).

File: docs/advanced-topics/docker-alternatives.rst [hunk 10]
```diff
@@ -73,6 +73,36 @@ add a line to your ``~/.dockstore/config``:
...
+::
+
+    cwltool-extra-parameters: --podman
+
+Cromwell
+~~~~~~~~
+For cromwell, we can set a symbolic link to run Podman instead of Docker on workflows.
```
Comment (aofarrel) [inline docs/advanced-topics/docker-alternatives.rst:99]: Nitpick: Cromwell should be capitalized.

**Review by aofarrel — CHANGES_REQUESTED** [review]
I think the first introductory paragraph needs a small change, which makes the rootless Docker split a little bit more jarring. Feel free to split into two PRs if more research is needed to properly combine the rootless Docker sections. Other than that, just some small nitpicks.

File: docs/advanced-topics/docker-alternatives.rst (hunk 3 above)
Comment (coverbeck) [reply docs/advanced-topics/docker-alternatives.rst:45]: Whichever way you go, I just noticed, it should be "have", not "has", to agree with "Multiple installations".

File: docs/advanced-topics/docker-alternatives.rst (hunk 2 above)
Comment (coverbeck) [reply docs/advanced-topics/docker-alternatives.rst:43]: I'm OK with this; IMHO it's deferring to whatever their  doc says.

File: docs/advanced-topics/docker-alternatives.rst [hunk 11]
```diff
@@ -73,6 +73,36 @@ add a line to your ``~/.dockstore/config``:
 
     cromwell-vm-options: -Dconfig.file=<absolute path to your Cromwell conf>
 
+Podman
+---------------
+
+Podman is a daemon-less alternative of Docker that allows end users to run docker/Open Container Initiative (OCI) containers without root privileges.
```
Comment (coverbeck) [inline docs/advanced-topics/docker-alternatives.rst:79]: docker -> Docker

File: docs/advanced-topics/docker-alternatives.rst (hunk 2 above)
Comment (hyunnaye) [reply docs/advanced-topics/docker-alternatives.rst:43]: I removed the comment and replaced the link to their current Singularity's install docs that the user can refer to. The old link was specifically for Singularity version 3.4 which is now outdated.

File: docs/advanced-topics/docker-alternatives.rst (hunk 4 above)
Comment (hyunnaye) [reply docs/advanced-topics/docker-alternatives.rst]: I can create a PR to add this to the dictionary. How is this definition of daemon? "Daemon is a background process that is used to create images and run containers. The Docker daemon constantly listens for Docker API requests and processes them."

**Review by aofarrel — APPROVED** [review]
Looks great! Thanks for doing this.

**Review by coverbeck — APPROVED** [review]

**Review by svonworl — APPROVED** [review]

**Review by kathy-t — APPROVED** [review]
